app/controllers/course/assessment/submissions_controller.rb

```diff
@@ -27,14 +27,26 @@ def edit
   end
 
   def update
+    if params[:submit_answer]
```

Comment:
It can't. Both guided and worksheet will need to submit a single answer ?  e.g. submit a coding question in worksheet for getting the auto grading result.